src/AutomatedCar/Models/AutomatedCar.cs

```diff
@@ -117,12 +120,35 @@ public void CalculateNextPosition()
         {
             double gasInputForce = this.gasPedalPosition * PEDAL_INPUT_MULTIPLIER;
             double brakeInputForce = this.brakePedalPosition * PEDAL_INPUT_MULTIPLIER;
-            double slowingForce = this.Speed * DRAG + (this.Speed > 0 ? brakeInputForce : 0);
 
-            this.Acceleration.Y = gasInputForce;
-            this.Velocity.Y += -(this.Acceleration.Y - slowingForce);
-            this.Y += (int)this.Velocity.Y;
-            this.CalculateSpeed();
+            double slowingForce = Speed * DRAG + (Speed > 0 ? brakeInputForce : 0);
+            Acceleration.Y = gasInputForce;
+            GearHandlingForVelocity(slowingForce);
+            Y += (int)Velocity.Y;
+            CalculateSpeed();
+        }
+
+        private void GearHandlingForVelocity(double slowingForce)
```

Comment:
I'd rather have this method return a value, instead of changing the Velocity internally. Using this pattern where possible, makes the code less complex. And you could also name rename the method to something that better describes it's functionality.